vowpalwabbit/core/include/vw/core/metrics_manager.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.=
+
+#pragma once
+
+#include "metric_sink.h"
+#include "vw/core/learner_fwd.h"
+
+#include <cstdint>
+#include <functional>
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+#include <vector>
+
+namespace VW
+{
+class metrics_manager
```

Comment:
if we remove the filename this could be renamed to `metrics_collector` and restrict the object from potentially becoming another all object in the future

Reply:
removed filename and updated to `metics_collector`